maptivist/src/main/java/com/google/sps/servlets/FlagServlet.java

```diff
@@ -0,0 +1,116 @@
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import com.google.sps.Marker;
+import java.util.Set;
+import java.util.List;
+import java.util.Base64;
+
+
+/** Servlet that handles all my received marker data */
+@WebServlet("/flags")
+public final class FlagServlet extends HttpServlet {
+    
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+ 
+        Gson gson = new Gson();
+ 
+        response.setContentType("application/json;");
+        response.getWriter().println(gson.toJson(Globals.flags));
+    }
+ 
+ 
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String titleMatch = request.getParameter("title");
+        String flag = request.getParameter("flag");
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        updateFlags(datastore, titleMatch, flag);
+
+        response.sendRedirect("/index.html");
+    }
+ 
+
+    public Entity getEntity(DatastoreService datastore, String titleMatch) {
+        Query query = new Query("Marker"); 
+        query.addFilter("title", FilterOperator.EQUAL, titleMatch); 
+        PreparedQuery pq = datastore.prepare(query);    
+        Entity entity = pq.asSingleEntity();
+
+        return entity;
+    }
+
+    public void updateFlags(DatastoreService datastore, String titleMatch, String flag) {
+        Entity entity = getEntity(datastore, titleMatch);
+        ArrayList<String> flags = createFlagObject((String) entity.getProperty("flags"));
+
+        flags.add(flag);
+        Globals.flags = new ArrayList<>(flags);
+
+        String flagString = createFlagString(flags);
+        entity.setProperty("flags", flagString);
+
+        datastore.put(entity);        
+    }
+
+    public String createFlagString(ArrayList<String> flags) {
+        String flagString = "";
+        Base64.Encoder encoder = Base64.getEncoder();  
+        for (String flag : flags) {
+            String comment = encoder.encodeToString(flag.getBytes());  
+            flagString += comment + ",";
+        }
+        return flagString.substring(0, flagString.length() - 1);
+    }
+ 
+    public ArrayList<String> createFlagObject(String flagString) {
+        Base64.Decoder decoder = Base64.getDecoder();  
+        ArrayList<String> flagsList = new ArrayList<String>();
+        String[] flags;
+        flags = flagString.split(",");
+        for (String flag : flags) {
+            String flagDecode = new String(decoder.decode(flag));
+            flagsList.add(flagDecode);
+        }
+        return flagsList;
+    }
+    
+    static class Globals {
+        public static ArrayList<String> flags;
```

Comment:
I thought each Marker was going to have its own set of flags? Why is this a global then?